lib/crowbar/client/app/database.rb

```diff
@@ -0,0 +1,220 @@
+
+        method_option :db_password,
+          type: :string,
+          default: "crowbar",
+          banner: "<db_password>",
+          desc: "Password for the Crowbar database password
+                                   Min length: 4
+                                   Max length: 63
+                                   Alphanumeric and special characters
+                                   Must begin with any alphanumeric character or underscore"
+
+        #
+        # Database create command
+        #
+        # it will set up a new Crowbar database before Crowbar installation.
+        # This command only works if Crowbar is not yet installed.
+        #
+        # @return [String] a formatted response from the server
+        #
+        def create
+          Command::Database::Create.new(
+            *command_params
+          ).execute
+        rescue => e
+          catch_errors(e)
+        end
+
+        desc "connect",
+          "Connect to a remote Crowbar database"
+
+        long_desc <<-LONGDESC
+          `connect` will connect to a remote Crowbar database before Crowbar installation.
+          This command only works if Crowbar is not yet installed.
+        LONGDESC
+
+        method_option :db_username,
+          type: :string,
+          default: "crowbar",
+          banner: "<db_username>",
+          desc: "Username for the Crowbar database user
+                                   Min length: 4
+                                   Max length: 63
+                                   Only alphanumeric characters and/or underscores
+                                   Must begin with a letter [a-zA-Z] or underscore"
+
+        method_option :db_password,
+          type: :string,
+          default: "crowbar",
+          banner: "<db_password>",
+          desc: "Password for the Crowbar database password
+                                   Min length: 4
+                                   Max length: 63
+                                   Alphanumeric and special characters
+                                   Must begin with any alphanumeric character or underscore"
+
+        method_option :database,
+          type: :string,
+          default: "crowbar_production",
+          banner: "<database>",
+          desc: "Name of the Crowbar database
+                                   Min length: 4
+                                   Max length: 63
+                                   Alphanumeric and special characters
+                                   Must begin with any alphanumeric character or underscore"
+
+        method_option :host,
+          type: :string,
+          default: "localhost",
+          banner: "<host>",
+          desc: "Host of the Crowbar database
+                                   Min length: 4
+                                   Max length: 63
+                                   Numbers and period characters (only IPv4)
+                                   Hostnames:
```

Comment:
this could also be FQDN which would include dot in allowed chars and max length = 253 (https://en.wikipedia.org/wiki/Hostname#Restrictions_on_valid_hostnames)